driver/src/bin/fetch_db.rs

```diff
@@ -0,0 +1,87 @@
+#[macro_use]
+extern crate serde_derive;
+
+mod models;
+
+extern crate serde_json;
+extern crate serde;
+extern crate mongodb;
+use std::io;
+use mongodb::bson;
+use mongodb::{Client, ThreadedClient};
+use mongodb::db::ThreadedDatabase;
+
+fn get_current_balances() -> Result<models::State, io::Error>{
```

Comment:
In order to unit test the components that use these methods later, you might have to put these methods into a trait that can be mocked.